src/vaults.lua

```diff
@@ -139,18 +139,22 @@ end
 -- return any vaults to owners that have expired
 function vaults.pruneVaults(currentTimestamp)
 	local allVaults = vaults.getVaults()
-	for owner, vaults in pairs(allVaults) do
-		for id, nestedVault in pairs(vaults) do
+	local prunedVaults = {}
+	for owner, ownersVaults in pairs(allVaults) do
+		for id, nestedVault in pairs(ownersVaults) do
 			if currentTimestamp >= nestedVault.endTimestamp then
 				balances.increaseBalance(owner, nestedVault.balance)
-				vaults[id] = nil
+				ownersVaults[id] = nil
+				prunedVaults[id] = nestedVault
 			end
 		end
 		-- update the owner vault
-		allVaults[owner] = vaults
+		-- TODO: I THINK THIS LINE IS UNNECESSARY. CHECK TESTS
+		allVaults[owner] = ownersVaults
```

Comment:
it's not - as we are updating the objects in place above, so not necessary - this was likely leftover from when we ported it over from smartweave

Reply:
Excised and tests ran fine:
be82619